### ISSUE TITLE
Show: [@printer] not considered in variants

### ISSUE THREAD

**Issue by johanneskloos** [whitequark/ppx_deriving#77]
Consider the following code (printer_variant.ml):
```
type t =
  | First [@printer fun fmt _ -> Format.fprintf fmt "first case"]
  | Second
      [@@deriving show]

let () =
  Format.printf "%a, %a@." pp First pp Second

```

Extrapolating from the documentation, I expect the output to be "first case, Printer_variant.Second". But:
```
$ ./printer_variant 
Printer_variant.First, Printer_variant.Second
```

Would it be possible to support adding [@printer] attributes to the cases of a variant type, so that the behavior matches my expectations?

**Comment by whitequark** [comment]
Sure, it's a straightforward improvement. You're welcome to write a PR.

**Comment by whitequark** [comment]
Fixed by #78.
